fastlane_core/lib/fastlane_core/configuration/config_item.rb

```diff
@@ -102,14 +102,12 @@ def initialize(key: nil, env_name: nil, description: nil, short_option: nil, def
       @allow_shell_conversion = (type == :shell_string)
     end
 
-    # This will raise an exception if the value is not valid
     def verify!(value)
-      UI.user_error!("Invalid value '#{value}' for option '#{self}'") unless valid?(value)
-      true
+      true if valid?(value)
```

Comment:
I think I find this a little bit confusing, I think - we have a method ending with a "!" that doesn't explicitly have a raise in it, and a method ending with a "?" that returns either true or else raises an exception.

Feels like the entire body of `valid?` should be moved into `verify!`, and the `true` return value should be eliminated (since it is always true if not raising).

Or am I missing something?  😄